### PR TITLE
 logging support in monkey patches

### PR DESCRIPTION
**Pull request by prabhakhar**
This has support for logging in "patch body". If the pattern is right, then I will make changes for the other patches.

### CODE REVIEW

File: modules/engine/lib/engine/source/verb.js [hunk 1]
```diff
@@ -378,6 +379,34 @@ var Verb = module.exports = function(table, statement, type, bag, path) {
             }
         });
     };
+
+    // Curry function for money patch logging
+    this.curry = function(log) {
```
Comment (prabhakhar) [inline modules/engine/lib/engine/source/verb.js:393]: Is the name curry OK?

File: modules/engine/lib/engine/source/verb.js [hunk 2]
```diff
@@ -580,7 +609,7 @@ function send(verb, args, uri, params, callback) {
     // Body
     var body;
     if(args.resource.method === 'post' || args.resource.method === 'put' || args.resource.method === 'delete' || args.resource.method === 'patch') {
-        var payload = args.resource.tmpl(parsed, params, headers, args.serializers);
+        var payload = args.resource.tmpl(parsed, params, headers, args.serializers, args);
```
Comment (prabhakhar) [inline modules/engine/lib/engine/source/verb.js]: We can pass "args.logEmitter" and "args.parentEvent". I prefer lesser no of parameters.

other way is to pass something like { 'emitter' : args.logEmitter, 'event' : args.parentEvent}.. but this too verbose.

Comment (s3u) [inline modules/engine/lib/engine/source/verb.js]: But why send both args.serializers, args?

**Comment by prabhakhar** [comment]
Do not merge yet. I need to push changes for other patches.

File: modules/engine/lib/engine/source/verb.js (hunk 1 above)
Comment (s3u) [inline modules/engine/lib/engine/source/verb.js:393]: Sure.

**Comment by prabhakhar** [comment]
Ready for merge. Please review and merge.

**Comment by s3u** [comment]
I published the modules to npm. make clean install will pull new ones.

@prabhakhar - could you add a note to http://ql.io/docs/monkey-patch on how to use this?

**Pull request merged**